src/commands/preview/dev_server.rs

```diff
@@ -0,0 +1,190 @@
+use std::net::{SocketAddr, ToSocketAddrs};
+
+use chrono::prelude::*;
+
+use hyper2::client::{HttpConnector, ResponseFuture};
+use hyper2::header::{HeaderName, HeaderValue};
+use hyper2::service::{make_service_fn, service_fn};
+use hyper2::{Body, Client, Request, Server, Uri};
+
+use hyper_tls2::HttpsConnector;
+
+use failure::format_err;
+
+use uuid::Uuid;
+
+use url::Url;
+
+use crate::commands;
+use crate::commands::preview::upload;
+
+use crate::settings::global_user::GlobalUser;
+use crate::settings::target::Target;
+
+use crate::terminal::emoji;
+
+const PREVIEW_HOST: &str = "rawhttp.cloudflareworkers.com";
+
+#[derive(Clone)]
+struct ServerConfig {
+    host: String,
+    listening_address: SocketAddr,
+    is_https: bool,
+}
+
+impl ServerConfig {
+    pub fn new(
+        host: Option<&str>,
+        ip: Option<&str>,
+        port: Option<&str>,
+    ) -> Result<Self, failure::Error> {
+        let port = port.unwrap_or("8000");
+
+        let try_address = match ip {
+            Some(ip) => format!("{}:{}", ip, port),
+            None => format!("localhost:{}", port),
+        };
+
+        let mut address_iter = try_address.to_socket_addrs()?;
```

Comment:
this seems somewhat arcane; does it make sense to pull it out into a named function?